src/main/java/com/parsing/parsers/pdf/parsing/RectangleDetector.java

```diff
@@ -0,0 +1,231 @@
+                x -> Imgproc.line(verticalLinesMat, new Point(x.getLeftPoint(), x.getYCoordinate()), new Point(x.getRightPoint(), x.getYCoordinate()),
+                        new Scalar(0, 0, 255), 3, Imgproc.LINE_AA, 0));
+        Imgcodecs.imwrite("All_Lines.png", verticalLinesMat);
+    }
+
+    private List<Double> mergeLines(List<Double> pointsY) {
+        log.debug("Lines has rows at start= " + pointsY.size());
+        List<Double> result = new ArrayList<>(pointsY);
+        for (int i = 0; i < pointsY.size(); i++) {
+            if (i + 1 < pointsY.size()) {
+                double point1 = pointsY.get(i);
+                double point2 = pointsY.get(i + 1);
+                double diff = point2 - point1;
+                if (diff < OFFSET) {
+                    result.remove(point2);
+                }
+            }
+        }
+        log.debug("Lines has rows at finished= " + result.size());
+        return result;
+    }
+
+    private List<Double> sortLinesByY(List<double[]> lines) {
+        List<Double> pointsY = new ArrayList<>();
+        for (double[] line : lines) {
+            pointsY.add(line[1]);
+        }
+        return pointsY.stream().sorted().toList();
+    }
+
+    private double[] sortListArray(List<double[]> lines) {
+        double minY = 50;
+        double maxY = 0;
+        double averageX = 0;
+        for (double[] doubles : lines) {
+            if (minY > doubles[1]) {
+                minY = doubles[1];
+            } else if (maxY < doubles[1]) {
+                maxY = doubles[1];
+            }
+            if (minY > doubles[3]) {
+                minY = doubles[3];
+            } else if (maxY < doubles[3]) {
+                maxY = doubles[3];
+            }
+            averageX = (averageX + doubles[0]) / 2;
+        }
+        return new double[]{averageX, maxY, averageX, minY};
+    }
+
+    private List<Double> sortLinesByX(List<double[]> lines) {
+        List<Double> result = new ArrayList<>();
+        for (double[] line : lines) {
+            result.add(line[0]);
+        }
+        return result.stream().sorted().toList();
+    }
+
+    private HorizontalLineCoordinate findExtremeHorizontalTablePoints(List<double[]> lines) {
+        double x1 = 100;
+        double x2 = 0;
+        for (double[] array : lines) {
+            if (array[0] < x1) {
+                x1 = array[0];
+            }
+            if (array[2] > x2) {
+                x2 = array[2];
+            }
+        }
+        return new HorizontalLineCoordinate(x1, x2, 0);
+    }
+
+    public List<double[]> findVerticalLinesWithOpenCV(String fileSource) {
+        Mat dst = new Mat();
+        Mat cdst = new Mat();
+        OpenCV.loadLocally();
+        Mat source = Imgcodecs.imread(fileSource, Imgcodecs.IMREAD_GRAYSCALE);
+        if (source.empty()) {
+            log.warn("Error opening image! Program Arguments: [image_name -- default " + fileSource + "] \n");
+            System.exit(-1);
```

Comment:
Why do we do Syste.exit(-1) ? Should we use it?

Reply:
Yes. I deleted it.